src/core/src/validation_util.cpp

```diff
@@ -151,7 +151,9 @@ namespace util {
 using ov::op::v0::Constant;
 
 std::shared_ptr<Constant> get_constant_from_source(const ov::Output<ov::Node>& source) {
-    if (const auto& c = ov::as_type_ptr<Constant>(source.get_node_shared_ptr())) {
+    if (!source.get_node()) {
```

Comment:
Why is required?

Reply:
If empty output it will create segfault.

Reply:
Ok, 
It looks like there are two issues:
- in `ov::as_type_ptr` which not handle nullptr correctly 
- in `has_and_set_equal_bounds` to handle null output properly

Could you add simple test in `src/core/tests/validation_utils.cpp` which reproduce this segfault without this change.

Reply:
Done